tools/translate.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/sh
+# Shell script helper for translating Metamath tool files to bash
+# Mario Carneiro 7-Jan-2022
+
+usage() {
+  cat << HELP
+TRANSLATE - Translate a TOOLS script into a corresponding bash script
+
+This command will take a script file intended for use with metamath
+tools-mode, and translate it into a bash script with the same behavior.
+See also SUBMIT. The script is passed the location of the tools
+directory (containing the core script files), which is also the location
+of this file. It uses 'mm-tools' by default, so if you symlink 'mm-tools'
+to the tools directory then these scripts will work.
+
+If the -r (recursive) option is passed, the generated file will translate
+SUBMIT calls to also make use of the same TOOLS directory; otherwise
+it will leave them off (using the default "mm-tools" directory).
+
+If the -e (echo) option is passed, the generated file will contain an
+echo statement for each executed command.
+
+Syntax:  translate.sh [-r] [-e] [TOOLS] < INFILE > OUTFILE
+HELP
+}
+
+if [ $# -eq 1 ] && [ "$1" = "-h" -o "$1" = "--help" ]; then usage; exit; fi
+
+if [ "$1" = "-r" ]; then recursive=1; shift; fi
+if [ "$1" = "-e" ]; then echo=1; shift; fi
+
+if [ $# -eq 0 ]; then tools="mm-tools"
+elif [ $# -eq 1 ]; then tools=$1
+else usage; exit 1; fi
+
+echo "#!/bin/bash"
+awk -v tools="$tools" -v echo="$echo" -v recursive="$recursive" \
```

Comment:
compile error
cat xx
exit
wolf@wolf-lammen:~/metamath/src/review/sts/tools$ ./translate.sh "/home/wolf/metamath/src/review/sts/tools" < xx > yy
awk: line 15: syntax error at or near {
awk: line 17: syntax error at or near :
awk: line 18: syntax error at or near :
awk: line 19: syntax error at or near :
awk: line 20: syntax error at or near :

Reply:
Hm, I can't replicate. Based on the position of the error, it sounds like your awk version doesn't support `switch`? If you can confirm, I'll rewrite it to avoid that.

Reply:
awk -W version
mawk 1.3.4 20200120
Copyright 2008-2019,2020, Thomas E. Dickey
Copyright 1991-1996,2014, Michael D. Brennan

random-funcs:       srandom/random
regex-funcs:        internal
compiled limits:
sprintf buffer      8192
maximum-integer     2147483647

Reply:
It seems it does not support a switch statement. This won't compile:
awk -v tools="$tools" -v echo="$echo" -v recursive="$recursive" \
'{
    x = $0
    switch (x)
    {
      case "x":
        print
    }
}'